Assets/Scripts/UI/PickupPup/ShopScreen/ShopItemSlot.cs

```diff
@@ -75,8 +75,14 @@ public void Init(ShopItem item)
     public void Buy()
     {
         gameController.TryBuyItem(item);
-    }
 
+        // If buying dogfood then update the amount of the appropriate DogFoodType
+        if ((int)item.ValueCurrencyType == 1)
```

Comment:
Should directly check for `CurrencyType.DogFood` equality instead of casting to an `int`, more readable